torch/testing/_internal/schema_check_mode.py

```diff
@@ -37,9 +44,17 @@ def has_mutated(before, after):
         for argument in func._schema.arguments:
             name = argument.name if argument.name != "self" else "input"
             if arguments.get(name) is not None:
-                before = tree_flatten(arguments.get(name))[0]
-                after = tree_flatten(cloned_arguments.get(name))[0]
+                before = tree_flatten(cloned_arguments.get(name))[0]
+                after = tree_flatten(arguments.get(name))[0]
                 if (any([has_mutated(i, j) for i, j in zip(before, after)]) and not argument.is_mutable):
                     raise RuntimeError(f"Argument {name} is not defined as mutable but was mutated")
+                for i in before:
+                    if not isinstance(out, tuple):
+                        if (have_values_aliased(i, out) and not is_aliasing(argument, func._schema.returns[0])):
+                            raise RuntimeError(f'Argument {name} is not defined to alias output but was aliasing')
```

Comment:
I think you should remove this check, ops are sometimes marked as aliasing when they only alias *sometimes*.

For example, `aten::to`.

```python
x = torch.rand((2, 2), dtype=torch.float)
y = x.to(dtype=torch.float)
z = x.to(dtype=torch.half)
```

^ in this example, I would expect that x and y are aliasing, but x and z are not aliasing.

Alternative option is to log when this happens but not error out, that way you might be able to identify ops that say they are aliasing but actually *never* alias.

Reply:
From what I understand, this check currently works for ops that are marked as aliasing when they only alias sometimes because it only raises an error if two values are aliasing **and** the arguments are expected to not be aliasing. 

For the case you mentioned above, the fact that x and z are not aliasing would not raise an error since the values are not aliasing and an error is only raised when values are aliasing. Though, I could definitely log out these other cases as you mentioned. If I wanted to do that what would I use to log?

Reply:
Whoops! Sorry, misread this line (thought it was the opposite check) - it looks good.